packages/core/src/entity/ArrayCollection.ts

```diff
@@ -80,10 +88,14 @@ export class ArrayCollection<T, O> {
    */
   hydrate(items: T[]): void {
     this.items.clear();
+    if (Utils.isDefined(this._count)) {
+      this._count = 0;
+    }
```

Comment:
i would prefer raw `typeof` instead of the `Utils` helper (in this particular file), as it could result in circular requires easily.

but here the condition does not make sense anyway, you would be re-adding to the cached value every time its re-hydrated

```suggestion
    this._count = 0;
```